pkg/api/types.go

```diff
@@ -799,6 +799,22 @@ type NodeCondition struct {
 	Message            string            `json:"message,omitempty"`
 }
 
+type NodeAddressKind string
+
+// These are valid address type of node. NodeLegacyHostIP is used to transit from out-dated
+// HostIP field to NodeAddress.
+const (
+	NodeLegacyHostIP NodeAddressKind = "LegacyHostIP"
+	NodeHostName     NodeAddressKind = "Hostname"
```

Comment:
I think we should keep HostIP in the internal representation until we remove support for v1beta1 and v1beta2.

Perhaps we should just have separate lists for the others. That seems like it would make common use cases simpler, where there is probably an order of preference among the different kinds.

That said, how is the master going to know which address to use?